lib/modulr/resources/payments/payment.rb

```diff
@@ -20,10 +20,12 @@ def initialize(raw_response, attributes = {})
 
         private def parse_details(details)
           case details[:type]
-          when "PI_SEPA_INST", "PI_FAST"
+          when "PI_SECT", "PI_SEPA_INST", "PI_FAST", "PI_REV"
             Details::Incoming::General.new(nil, details)
-          else
+          when "PO_SECT", "PO_SEPA_INST", "PO_FAST", "PO_REV"
             Details::Outgoing::General.new(nil, details)
+          when nil
```

Comment:
Instead of using else clause we control that everytime we do not receive payment type we are hadle it as Internal.